### DISCUSSION TITLE
Icons representing active tab processes

### DISCUSSION THREAD

**Discussion by joncrangle** [michaelbrusegard/tabline.wez#7] [Show and tell]
Thanks @michaelbrusegard for creating this project.

Here's a show and tell for how I am using `tabline.wez` to show icons for the active tab processes in my tab bar:

![image](https://github.com/user-attachments/assets/1a7f0509-2e72-47d7-8a51-fa3f65725306)

Create a lua table with process names as keys and corresponding icons [ref: wezterm.nerdfonts](https://wezfurlong.org/wezterm/config/lua/wezterm/nerdfonts.htm). Make sure to include a `default` key and value since this will be a fallback icon.

```lua
local icons = {
  ['bash'] = wezterm.nerdfonts.cod_terminal_bash,
  ['btm'] = wezterm.nerdfonts.mdi_chart_donut_variant,
  ['btop'] = wezterm.nerdfonts.md_chart_areaspline,
  ['cargo'] = wezterm.nerdfonts.dev_rust,
  ['cmd.exe'] = wezterm.nerdfonts.md_console_line,
  ['curl'] = wezterm.nerdfonts.mdi_flattr,
  ['debug'] = wezterm.nerdfonts.cod_debug,
  ['default'] = wezterm.nerdfonts.md_application,
  ['docker'] = wezterm.nerdfonts.linux_docker,
-- ... etc
}
```

The following function looks for a key that starts with the name returned by the `process` component.
```lua
local function get_icon(str)
  for k, v in pairs(icons) do
    if str:lower():match("^" .. k) then
      return v
    end
  end
  return icons['default']
end
```

Within your `tabline.wez` setup for `tab_active` and/or `tab_inactive` use the `get_icon` function:
```lua
        {
          'process',
          fmt = function(str)
            return get_icon(str)
          end
        },
```

**Comment by michaelbrusegard** [comment]
As of https://github.com/michaelbrusegard/tabline.wez/pull/14, this is now a built in functionality 🎉
